### ISSUE TITLE
Suggestion: Cooling option

### ISSUE THREAD

**Issue by mrillies** [SergiuToporjinschi/node-red-contrib-heater-controller#23]
Hi, This widget looks quite powerful and well built.

It would be nice to have a cool or dual output function, making it similar to software version of an STC-1000. 

From what i can see would only require a second output (or an additional message so as not to break current compatibility) and a second set of hysterisis values. 

Thoughts or best to fork this to a seperate project?

**Comment by SergiuToporjinschi** [comment]
Hi, sorry for the late response... 
Let me see if I understood. You want, basically to set a temperature interval and keep the temp in that interval. Let's say you put an upper limit and a lower limit. 
Low: 20 and High; 25 
and if the current temperature is 18 then turn the heater on and the cooler off;
if the current temperature is above 25 like 28 then turn the cooler on and heater off;
if the current temperature is 23 then turn the heater and the cooler off;
I understood correctly?

**Comment by mrillies** [comment]
More like two back to back hysterisis.

Set point 20deg

Heat Hyst off= 1
Heat Hyst on =2

Cool Hyst off= 0
Cool Hyst on = 2

Start 20
Drop to 18 - heater turns on
Raise to 19 heater turns off
Natural rise to 22 cooler turns on
Drop to 20 cooler turns off

Basically it’s just a second copy of the hysterisis but inverted, and maybe
a parameter for “dead time” so it dosent switch on heat and cool too
quickly after the other

**Comment by SergiuToporjinschi** [comment]
I understood you want to have a single temperature and not an interval like I described. 
and with hysteresis. I would go on intervals like I proposed first time but with hysteresis for not killing the cooler and the heater.  And I would go like this because I see it more flexible and I think would work for you too if you are setting both low and high on the same value. I think;

The problem is that I do not have time know and I'm planning to refactor this project because I don't like my code that I wrote. In fact I never like my code after it passes some time :)... 

So, if you fork it I cannot grantee that I will be able to merge it after. and to merge it before refactoring is possible to be hard to refactor... 

But nevertheless is a great idea and I will implement it;
Meanwhile, maybe you can achieve  what you need by using same not twice in a workflow one for lowering temperature and one for rising it... I think you would have some work to there and is not user friendly ....

**Comment by mrillies** [comment]
Yeah I very much like your complete implementation, having two seperate
widgets is not ideal.

Based on your plans I will just fork and quick hack your current
implementation and come back to this later :)

Thanks for taking the time to discuss :)

**Comment by SergiuToporjinschi** [comment]
aa no problem, be my guest :) yah maybe I can merge it ...

**Comment by ahmaddxb** [comment]
What part of the code would need to be changed in order for this to work for cooling. I want it to turn on when the temperature is higher then the target value rather than lower as it work now.

**Comment by shortyishere** [comment]
Did we add this an option for it to switch heating or cooling or add it as one module? I would like the same
